### DISCUSSION TITLE
System.Exception: Shape type 'ShortcodeModal' not found

### DISCUSSION THREAD

**Discussion by PiemP** [OrchardCMS/OrchardCore#6908] [Q&A] [answered]
I have an issue on latest dev branch. 

If I try to publish the project `OrchardCore.Cms.Web` i receive this error: 

`System.Exception: Shape type 'ShortcodeModal' not found`

I already have do dotnet clean and dotnet build but seems doesn't solve the problem. 

Can someone confirm this?

**Comment by deanmarcussen** [comment, marked as answer]
We recased it, but you may have to reenable the feature.

https://docs.orchardcore.net/en/dev/docs/reference/modules/Shortcodes/#shortcodes

**Reply by PiemP** [reply]
Thank you @deanmarcussen ... I presume it's a stupid question 😅 but, if someone doesn't want the Shortcode feature, what he have to do? Seems the code is executed in the `HtmlBodyPart-Trumbowyg.Edit.cshtml` without check if the feature is enabled or not.

**Reply by deanmarcussen** [reply]
The Html / Markdown features depend on `Shortcodes` - we need it to be able to supply the image shortcode from.

So the feature was actually running, but the something has issues with the case sensitivity of the feature name, which changed from `ShortCodes` to `Shortcodes`, so it doesn't find the shape.

Shortcode Templates are a feature so you can turn them off. 

We could have a null provider if it's a major issue having the feature, but it makes the code much more complicated.

**Reply by PiemP** [reply]
I asked this because I don't have the feature enabled in this Orchard Core instance and seems strange that I have an error about a feature not enabled.

In any case, thank you @deanmarcussen for your super fast answer.
